File: JPAChildcare/src/main/java/com/skilldistillery/childcare/entities/Staff.java
package com.skilldistillery.childcare.entities;

import java.util.Objects;

import javax.persistence.Column;
import javax.persistence.Entity;

@Entity
public class Staff {
	
	@Column(name = "first_name")
	private String firstName;
	@Column(name = "last_name")
	private String lastName;
	
	@Column(name="image_url")
	private String imageUrl;
	//user
	//classroom

	@Override
	public String toString() {
		return "Staff [firstName=" + firstName + ", lastName=" + lastName + ", imageUrl=" + imageUrl + "]";
	}

	public String getFirstName() {
		return firstName;
	}

	public void setFirstName(String firstName) {
		this.firstName = firstName;
	}

	public String getLastName() {
		return lastName;
	}

	public void setLastName(String lastName) {
		this.lastName = lastName;
	}

	public String getImageUrl() {
		return imageUrl;
	}

	public void setImageUrl(String imageUrl) {
		this.imageUrl = imageUrl;
	}

	@Override
	public int hashCode() {
		return Objects.hash(firstName, imageUrl, lastName);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Staff other = (Staff) obj;
		return Objects.equals(firstName, other.firstName) && Objects.equals(imageUrl, other.imageUrl)
				&& Objects.equals(lastName, other.lastName);
	}
	

}

File: ngDaycare/src/app/models/staff.ts
import { Classroom } from "./classroom";
import { User } from "./user";

export class Staff {
  id: number;
  firstName: string | undefined;
  lastName: string | undefined;
  imageUrl: string | undefined;
  classroom: Classroom | undefined;
  user: User | undefined;

  constructor(id: number = 0, firstName?: string, lastName?: string, imageUrl?: string, classroom?: Classroom, user?: User){
    this.id = id;
    this.firstName = firstName;
    this.lastName = lastName;
    this.imageUrl = imageUrl;
    this.classroom = classroom;
    this.user = user;
  }
}
